File: backend/models/materi_pembelajaran.go
package models

import "time"

type MateriPembelajaran struct {
	ID          int       `json:"id" db:"id"`
	MentorID    int       `json:"mentor_id" db:"mentor_id"`
	Title       string    `json:"title" db:"title"`
	Description string    `json:"description" db:"description"`
	FilePath    string    `json:"file_path" db:"file_path"`
	FileType    string    `json:"file_type" db:"file_type"`
	FileSize    int64     `json:"file_size" db:"file_size"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

File: frontend_website_bmc/lib/models/materi_pembelajaran.dart
class MateriPembelajaran {
  final int id;
  final int mentorId;
  final String classLevel;
  final String title;
  final String description;
  final String filePath;
  final String fileType;
  final int fileSize;
  final DateTime createdAt;
  final DateTime updatedAt;

  MateriPembelajaran({
    required this.id,
    required this.mentorId,
    required this.classLevel,
    required this.title,
    required this.description,
    required this.filePath,
    required this.fileType,
    required this.fileSize,
    required this.createdAt,
    required this.updatedAt,
  });

  factory MateriPembelajaran.fromJson(Map<String, dynamic> json) {
    final kelas =
        (json['kelas'] ??
                json['class_level'] ??
                json['class'] ??
                json['paket'] ??
                json['paket_nama'] ??
                json['class_name'])
            ?.toString() ??
        '';
    return MateriPembelajaran(
      id: json['id'] ?? 0,
      mentorId: json['mentor_id'] ?? 0,
      classLevel: kelas,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      filePath: json['file_path'] ?? '',
      fileType: json['file_type'] ?? '',
      fileSize: json['file_size'] ?? 0,
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
      updatedAt: DateTime.tryParse(json['updated_at'] ?? '') ?? DateTime.now(),
    );
  }
}
